tests/test_football_parser.py

```
@@ -0,0 +1,173 @@
+import pytest
+import os
+import shutil
+import requests
+
+from src.football_parser import FootballParser
+
+HOSTNAME = "http://test.com"
+TMP_FOLDER = "./tmp"
+
+
+@pytest.fixture
+def fp():
+    fp = FootballParser(HOSTNAME + "/test.php", TMP_FOLDER)
+
+    return fp
+
+
+@pytest.fixture
+def use_tmp_folder():  # Fixture to create and remove the temp file folder
+    assert not os.path.isdir(TMP_FOLDER)
+
+    yield
+
+    shutil.rmtree(TMP_FOLDER, ignore_errors=True)
+
+
+@pytest.mark.parametrize(
+    "season_text",
+    [
+        None,
+        "",
+        "season 2019/2020",
+        "Season 2019/202",
+        " Season 2019/2020",
+        "Season 3019/2020",
+        "<i>Season 2019/2020</i>",
+    ],
+)
+def test_no_new_season(fp, season_text):
+    fp._check_new_season(season_text)
+    assert fp._curr_season is None
+
+
+@pytest.mark.parametrize("season_text", ["Season 2019/2020", "Season 1000/2999"])
+def test_new_season_ok(fp, season_text):
+    fp._check_new_season(season_text)
+    assert fp._curr_season == season_text.split(" ")[-1]
+
+
+@pytest.mark.parametrize(
+    "tag,text,href",
+    [
+        (None, "Test League 1", "test.csv"),
+        ("a", None, "test.csv"),
+        ("a", "Test League 1", None),
+    ],
+)
+def test_save_data_with_incomplete_data(fp, tag, text, href):
+    fp._curr_season = "2019/2020"
+    fp._save_data(tag, text, href)
+    assert not os.path.isdir(TMP_FOLDER)
```

Instead of silently failing and not saving the files, in case of errors the `_save_data()` method should raise an exception describing the correct type of the error.